.github/workflows/build.yaml

```diff
@@ -0,0 +1,149 @@
+name: ci
+
+on:
+  push:
+    branches: main
+
+  # Setting this enables manually triggering workflow in the GitHub UI
+  # see https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow
+  workflow_dispatch: {}
+
+permissions: read-all
+
+# Two probers running at once can break each other.
+concurrency:
+  group: builders
+  cancel-in-progress: true
+
+jobs:
+  build_prod:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Run build-prod civiform:latest
+        id: build_and_push
+        env:
+          DOCKER_BUILDKIT: 1
+          PUSH_IMAGE: 1
+          DOCKER_HUB_ACCESS_TOKEN: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
+          DOCKER_HUB_USERNAME: ${{ secrets.DOCKER_HUB_USERNAME }}
+        run: bin/build-prod
+
+  build_dev:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Build and push new development image to docker hub
+        id: build_dev
+        env:
+          DOCKER_BUILDKIT: 1
+          PUSH_IMAGE: 1
+          DOCKER_HUB_ACCESS_TOKEN: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
+          DOCKER_HUB_USERNAME: ${{ secrets.DOCKER_HUB_USERNAME }}
+        run: bin/build-dev
+
+  build_formatter:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - id: file_change
+        uses: trilom/file-changes-action@v1.2.4
+        if: ${{ github.event_name == 'push' }}
```

Comment:
issue (blocking): here and with the other jobs. the file_change step only runs if the event is push, but the later step that depends on output from file_change runs if the event is not push

Reply:
The later step is `        if: ${{ github.event_name != 'push' }} ||  contains(toJSON(steps.file_changes.outputs.files), 'formatter/')`

AKA, if it's not a push event (it was manually triggered), always build the container

Reply:
Added a comment